src/daft-dsl/src/expr.rs

```diff
@@ -568,6 +569,86 @@ impl Expr {
             _ => None,
         }
     }
+
+    pub fn to_sql(&self) -> Option<String> {
+        fn to_sql_inner<W: Write>(expr: &Expr, buffer: &mut W) -> io::Result<()> {
+            match expr {
+                Expr::Column(name) => write!(buffer, "{}", name),
+                Expr::Literal(lit) => {
+                    if let Some(s) = lit.to_sql() {
+                        write!(buffer, "{}", s)
+                    } else {
+                        Err(io::Error::new(
+                            io::ErrorKind::Other,
+                            "Unsupported literal for SQL translation",
+                        ))
+                    }
+                }
+                Expr::Alias(inner, ..) => to_sql_inner(inner, buffer),
+                Expr::BinaryOp { op, left, right } => {
+                    to_sql_inner(left, buffer)?;
+                    let op = match op {
+                        Operator::Eq => "=".to_string(),
+                        Operator::NotEq => "!=".to_string(),
+                        Operator::Lt => "<".to_string(),
+                        Operator::LtEq => "<=".to_string(),
+                        Operator::Gt => ">".to_string(),
+                        Operator::GtEq => ">=".to_string(),
+                        Operator::And => "AND".to_string(),
+                        Operator::Or => "OR".to_string(),
+                        _ => {
+                            return Err(io::Error::new(
+                                io::ErrorKind::Other,
+                                "Unsupported operator for SQL translation",
+                            ))
+                        }
+                    };
+                    write!(buffer, " {} ", op)?;
+                    to_sql_inner(right, buffer)
+                }
+                Expr::Not(inner) => {
+                    write!(buffer, "NOT (")?;
```

Comment:
you should be able to collapse this 
`write!(buffer, "NOT ({})", o_sql_inner(inner, buffer)?)`